internal/circle/errors.go

```diff
@@ -28,15 +32,33 @@ func (e APIError) Error() string {
 	return fmt.Sprintf("APIError: Code=%d, Message=%s, Errors=%v", e.Code, e.Message, e.Errors)
 }
 
+type TenantStatusUpdater struct {
+	tenantManager tenant.ManagerInterface
+}
+
+var invalidAPIKeyStatusCodes = []int{http.StatusUnauthorized, http.StatusForbidden}
+
 // parseAPIError parses the error response from Circle APIs.
 // https://developers.circle.com/circle-mint/docs/circle-apis-api-errors.
-func parseAPIError(resp *http.Response) (*APIError, error) {
+func (u TenantStatusUpdater) parseAPIError(ctx context.Context, resp *http.Response) (*APIError, error) {
```

Comment:
Something with this `TenantStatusUpdater` feels wrong to me. I think we're mixing concerns here. 

This `errors.go` file should be just about defining and parsing errors coming back from circle. The logic of deactivating an account should be encapsulated elsewhere (client). 

We can add a method to the client, `handleApiError(ctx, resp)` that parses the error then handles it appropriately.